src/prpy/tsr/tsr.py

```diff
@@ -52,36 +61,130 @@ def __init__(self, T0_w=None, Tw_e=None, Bw=None,
             self.manipindex = manip
         self.bodyandlink = bodyandlink
 
-    def sample(self, vals=None):
-        Bwdims = 0
-        for i in range(6):
-            if self.Bw[i, 0] != self.Bw[i, 1]:
-                Bwdims += 1
-
-        if vals is None:
-            Bwvals = self.Bw[:, 0] + (self.Bw[:, 1] - self.Bw[:, 0]) \
-                * numpy.random.random_sample(6)
-        elif len(vals) == Bwdims:
-            Bwvals = numpy.zeros(6)
-            vals_i = 0
-            for i in range(6):
-                if self.Bw[i, 0] != self.Bw[i, 1]:
-                    Bwvals[i] = vals[vals_i]
-                    vals_i += 1
-                else:
-                    Bwvals[i] = self.Bw[i, 0]
-        elif len(vals) == 6:
-            Bwvals = vals
-        else:
-            raise ValueError('vals must be of length %d or 6!' % Bwdims)
-        # print 'Bwvals[5]:', Bwvals[5]
-        xyzypr = [Bwvals[0], Bwvals[1], Bwvals[2],
-                  Bwvals[5], Bwvals[4], Bwvals[3]]
-        Tw = kin.pose_to_H(kin.pose_from_xyzypr(xyzypr))
+    def to_transform(self, xyzrpy):
+        """
+        Converts a [x y z roll pitch yaw] into an
+        end-effector transform.
 
+        @param  xyzrpy [x y z roll pitch yaw]
+        @return trans 4x4 transform
+        """
+        if len(xyzrpy) != 6:
+            raise ValueError('vals must be of length 6')
+        if not self.is_valid(xyzrpy):
+            raise ValueError('Invalid xyzrpy')
+
+        xyzypr = [xyzrpy[0], xyzrpy[1], xyzrpy[2],
+                  xyzrpy[5], xyzrpy[4], xyzrpy[3]]
+        Tw = kin.pose_to_H(kin.pose_from_xyzypr(xyzypr))
         trans = numpy.dot(numpy.dot(self.T0_w, Tw), self.Tw_e)
```

Comment:
How do people feel about `trans = reduce(numpy.dot, [self.T0_w, Tw, self.Tw_e])` for things like this?